ChiTech/ChiPhysics/FieldFunction/fieldfunction_vtkpwld.cc

```diff
@@ -278,226 +36,75 @@ void chi_physics::FieldFunction::ExportToVTKPWLDG(const std::string& base_name,
-
+      component_names[c] = field_name + ff_uk.component_text_names[c];
+      field_node_array[c]->SetName(component_names[c].c_str());
+      field_cell_avg_array[c]->SetName((component_names[c] +
+                                        std::string("-avg")).c_str());
     }
 
-    //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% POLYGON
-    if (cell.Type() == chi_mesh::CellType::POLYGON)
-    {
-      auto poly_cell = (chi_mesh::CellPolygon*)(&cell);
-
-      int num_verts = poly_cell->vertex_ids.size();
-      std::vector<vtkIdType> cell_info(num_verts);
-      for (int v=0; v<num_verts; v++)
-      {
-        uint64_t vgi = poly_cell->vertex_ids[v];
-        std::vector<double> d_node(3);
-        d_node[0] = grid->vertices[vgi].x;
-        d_node[1] = grid->vertices[vgi].y;
-        d_node[2] = grid->vertices[vgi].z;
-
-        points->InsertPoint(nc,d_node.data());
-        cell_info[v] = nc; nc++;
-      }
-
-      ugrid->
-        InsertNextCell(VTK_POLYGON,num_verts,
-                       cell_info.data());
 
-      matarray->InsertNextValue(mat_id);
-      pararray->InsertNextValue(cell.partition_id);
+  //############################################# Populate cell information
+  int64_t node_count=0;
+  for (const auto& cell : grid->local_cells)
+  {
+    UploadCellGeometry(cell, node_count, points, ugrid);
 
-      //============= Create dof mapping
-      std::vector<uint64_t> mapping;
-      std::vector<std::tuple<uint64_t,uint,uint>> cell_node_component_tuples;
+    material_array->InsertNextValue(cell.material_id);
+    partition_id_array->InsertNextValue(cell.partition_id);
 
-      for (int g=0; g < ff_uk.num_components; g++)
-        for (int v=0; v<num_verts; v++)
-          cell_node_component_tuples.emplace_back(cell_local_id,v,g);
+    //Populate Arrays
+    size_t num_nodes = cell.vertex_ids.size();
+    std::vector<uint64_t> mapping;
+    std::vector<std::tuple<uint64_t,uint,uint>> cell_node_component_tuples;
 
-      CreatePWLDMappingLocal(cell_node_component_tuples,mapping);
+    for (int c=0; c < num_components; c++)
+      for (int v=0; v < num_nodes; v++)
+        cell_node_component_tuples.emplace_back(cell.local_id, v,
+                                                (num_components==1)?
+                                                ref_component : c);
 
-      int counter=-1;
-      for (int g=0; g < ff_uk.num_components; g++)
-      {
-        double cell_avg_value = 0.0;
-        for (int v=0; v<num_verts; v++)
-        {
-          ++counter;
-          double dof_value = field_vector_local->operator[](mapping[counter]);
-          cell_avg_value+= dof_value;
-          phiarray[g]->InsertNextValue(dof_value);
-        }
-        phiavgarray[g]->InsertNextValue(cell_avg_value/num_verts);
-      }//for g
-    }
+    CreatePWLDMappingLocal(cell_node_component_tuples, mapping);
 
-    //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% POLYHEDRON
-    if (cell.Type() == chi_mesh::CellType::POLYHEDRON)
+    int counter=-1;
```

Comment:
just wondering what this counter starts at -1.

Reply:
Changed to type `size_t` and changed logic to start at 0.